src/object.c

```diff
@@ -455,10 +455,15 @@ robj *tryObjectEncoding(robj *o) {
             incrRefCount(shared.integers[value]);
             return shared.integers[value];
         } else {
-            if (o->encoding == OBJ_ENCODING_RAW) sdsfree(o->ptr);
-            o->encoding = OBJ_ENCODING_INT;
-            o->ptr = (void*) value;
-            return o;
+            if (o->encoding == OBJ_ENCODING_RAW) {
+                sdsfree(o->ptr);
+                o->encoding = OBJ_ENCODING_INT;
+                o->ptr = (void*) value;
+                return o;
+            } else {
+                decrRefCount(o);
+                return createStringObjectFromLongLongForValue(value);
```

Comment:
acutally we can just call `return createObject(OBJ_STRING,sdsfromlonglong(value));`
`createStringObjectFromLongLongForValue` will do more work like `maxmemory` check and `shared.integers` check, those check we already did above

just a casually see, never mind